lib/SILOptimizer/Utils/ConstantFolding.cpp

```diff
@@ -1325,6 +1349,12 @@ static bool constantFoldInstruction(Operand *Op, Optional<bool> &ResultsInError,
                                     SmallVectorImpl<SILValue> &Results) {
   auto *User = Op->getUser();
 
+  // Diagnose potentially out of bounds subscript access
+  if (auto *AI = dyn_cast<ApplyInst>(User)) {
```

Comment:
You should be able to just pass in FullApplySite without creating an lvalue here. I.e.:

foo(FullApplySite(AI)).